assets/dev/js/editor/utils/controls-css-parser.js

```diff
@@ -247,6 +255,10 @@ ControlsCSSParser = elementorModules.ViewModule.extend( {
 			value = control.selectors_dictionary[ value ] || value;
 		}
 
+		if ( control.unit_selectors_dictionary && value.unit ) {
```

Comment:
What you are replacing here is the `{{VALUE}}` placeholder.
What I believe you should be replacing is the entire selector (the `outputCssProperty` contents inside `addControlStyleRules()`), before it goes through the regex replace process.

For example, let's say you add a control with support for `px`, `%`, `em`, `vw`, `custom`.
Say you want a custom selector _only for the `custom` units value_.
Using the current implementation, you would have to repeatedly define the same exact selector for all other unit types, like so:
```
'unit_selectors_dictionary' => [
	'px' => '--my-property: repeat({{SIZE}}, {{UNIT}})',
	'%' =>  '--my-property: repeat({{SIZE}}, {{UNIT}})',
	'em' => '--my-property: repeat({{SIZE}}, {{UNIT}})',
	'vw' => '--my-property: repeat({{SIZE}}, {{UNIT}})',
	'custom' => '--my-custom-property: {{SIZE}}',
],
```
This doesn't make sense.
Our implementation should allow for a default selector (defined in the `'selectors'` property, like normal), which is only overridden with the custom per-unit-type selector if the chosen unit exists in the unit selectors dictionary.

Replacing the entire `outputCssProperty`, instead of the control value, will allow developers to define a default selector like they do today under the `'selectors'` property, and then define the dictionary that will only override the selector for a specific unit choice.

If this isn't clear, let's talk about it more 1on1.

Reply:
@udidol You are correct. Thank you. This has been updated in the JS and the PHP.